art/attacks/evasion/adversarial_patch/adversarial_patch_tensorflow.py

```diff
@@ -335,47 +368,100 @@ def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> T
 
         :param x: An array with the original input images of shape NHWC or input videos of shape NFHWC.
         :param y: An array with the original true labels.
+        :param mask: An boolean array of shape equal to the shape of a single samples (1, H, W) or the shape of `x`
+                     (N, H, W) without their channel dimensions. Any features for which the mask is True can be the
+                     center location of the patch during sampling.
+        :type mask: `np.ndarray`
         :return: An array with adversarial patch and an array of the patch mask.
         """
         import tensorflow as tf  # lgtm [py/repeated-import]
 
-        y = check_and_transform_label_format(labels=y, nb_classes=self.estimator.nb_classes)
-
         shuffle = kwargs.get("shuffle", True)
-        if shuffle:
-            dataset = (
-                tf.data.Dataset.from_tensor_slices((x, y))
-                .shuffle(10000)
-                .batch(self.batch_size)
-                .repeat(math.ceil(x.shape[0] / self.batch_size))
+        mask = kwargs.get("mask")
```

Comment:
Line 380-388: I see this part of codes for processing mask is very similar to that part in the file adversarial_patch_numpy.py, is it possible to place this code in a common function?

Reply:
I agree, we should unify the checks across all attacks.